### ISSUE TITLE
AniList Import List imports many false positives

### ISSUE THREAD

**Issue by mousesama** [Sonarr/Sonarr#6140]
### Is there an existing issue for this?

- [X] I have searched the existing open and closed issues

### Current Behavior

Connecting an Import List from AniList to my Sonarr instances syncs up my series as expected but also adds several false positives, both anime series and completely unrelated TV shows, including ones from YouTube for some reason.
Example of series that were added and I had to exclude them: 
![chrome_lmiEw2WFOY](https://github.com/Sonarr/Sonarr/assets/68248807/444772fa-1e9a-4a73-9ea4-41885bb24e45)

### Expected Behavior

Only my anime series from AniList get imported, as defined in the config.

### Steps To Reproduce

Create an AniList Import List with a large amount of series. Here is the config I used:
![chrome_9emOKTFtjw](https://github.com/Sonarr/Sonarr/assets/68248807/c64d91ab-110a-4693-8222-b8e9b525f34d)
![chrome_2EBsLqi74u](https://github.com/Sonarr/Sonarr/assets/68248807/631a1f1d-2356-43b1-a69e-656876415f0e)

### Environment

```markdown
- OS: Unraid
- Sonarr: 4.0.0.710
- Docker Install: Yes
- Using Reverse Proxy: Yes
- Browser: Google Chrome 118
- Database: Sqlite 3.41.2
```

### What branch are you running?

Develop

### Trace Logs?

Will update with logs when the Import List sync triggers again. I have removed some previous exclusions from the exclusion list so they should get erroneously imported again.

### Anything else?

Profile that's being imported from: https://anilist.co/user/mousesama/

**Comment by augustuen** [comment]
You haven't defined that only Anime series should be imported, you've told Sonarr to add the series as Anime, which it'll do regardless of what kind of series it's adding.

**Comment by markus101** [comment]
The issue here is Sonarr ties to find a match by Anilist ID, which fails for some (because there isn't a mapping for everything). For example this item in the list:

Anilist ID: 2248
Title: Hidden Leaf Village Grand Sports Festival

No result by Anilist ID is found so it falls back to searching by the title, which returns 100 results, the first being `Genius Chef Junior`.

At least for Anilist we don't want to fall back to title lookups (for now that's the only one I'm going to change).
